components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs

```diff
@@ -507,6 +537,14 @@ private sealed class DataSetWriterSubscription : IDisposable {
             private long _currentSequenceNumber;
         }
 
+        /// <summary>
+        /// Helper to deserialize credential info
+        /// </summary>
+        private class cred {
```

Comment:
Can we make this into a public class and use it whenever we serialize/deserialize `CredentialModel.Value` for `CredentialType.UserName` case ? I can see 3 places now where we hardcode the expected structure:
* in `PublishedNodesJobConverter.ToUserNamePasswordCredentialAsync()` method
* in `StackModelsEx.ToServiceModel()` method
* and now here